File: Lab-4/fibo.py
table = [0] * 100
def fibo(n):
    if n==1 or n==2 :
        return 1
    else:
        if table[n]==0:
            table[n]=fibo(n-1)+fibo(n-2)
        return table[n]
    
n = int(input("Enter n:"))
print(f"The {n}th term is {fibo(n)}")

File: Lab-4/fibo.c
#include<stdio.h>
long int table[100]={0};
long int fibo(int n)
{
    if (n==1 || n==2)
        return 1;
    else{
        if (table[n]==0)
        {
            table[n]=fibo(n-1)+fibo(n-2);
        }
        return table[n];
    }
}
int main()
{
    int n;
    printf("Enter n:");
    scanf("%d",&n);
    printf("the %dth term is %d\n",n,fibo(n));
    return 0;
}